packages/astro/src/core/render/core.ts

```diff
@@ -35,6 +35,27 @@ export async function getParamsAndProps(
 			const paramsMatch = route.pattern.exec(pathname);
 			if (paramsMatch) {
 				params = getParams(route.params)(paramsMatch);
+
+				// If we have an endpoint at `src/pages/api/[slug].ts` that's prerendered, and the `slug`
+				// is `undefined`, throw an error as we can't generate the `/api` file and `/api` directory
+				// at the same time. Using something like `[slug].json.ts` instead will work.
+				if (route.type === 'endpoint' && mod.getStaticPaths) {
+					const lastSegment = route.segments[route.segments.length - 1];
```

Comment:
Could `lastSegment` be `undefined`? Do we have at least one segment here?

Reply:
@bluwy help check

Reply:
Yeah since we're handling a dynamic route, it would always have a segment in the path here, so it should not be `undefined`